### PR TITLE
Don't keep sessions open while doing multi-second operations

### PR DESCRIPTION
**Pull request by djmitche**

### CODE REVIEW

**Comment by djmitche** [comment]
@Callek r?

File: relengapi/blueprints/tooltool/grooming.py [hunk 1]
```diff
@@ -131,13 +134,16 @@ def check_pending_upload(session, pu):
...
+
     if time.now() < pu.expires:
         # URL is not expired yet
         return
     elif time.now() > pu.expires + timedelta(days=1):
         # Upload will probably never complete
         log.info(
-            "Deleting abandoned pending upload for {}".format(pu.file.sha512))
+            "Deleting abandoned pending upload for {}".format(sha512))
         session.delete(pu)
```
Comment (Callek) [inline relengapi/blueprints/tooltool/grooming.py]: with ```session.delete(pu)``` here, and the removal of ```session.commit()``` in ```check_pending_uploads``` don't we miss actually committing the delete's?

Comment (Callek) [inline relengapi/blueprints/tooltool/grooming.py]: err context folded out, ignore this one.

Comment (Callek) [inline relengapi/blueprints/tooltool/grooming.py]: ok, now with my thinking cap on correctly, the return on the next line makes *this* comment valid.

**Comment by Callek** [comment]
merely skimmed the tests and they look good, but as noted above I think the badpenny tasks's final commit() needs to exist. (*or* we should commit at the point where we delete, but before we return)

**Comment by Callek** [comment]
:+1: with that addressed (no need for a second look)

File: relengapi/blueprints/tooltool/grooming.py (hunk 1 above)
Comment (djmitche) [inline relengapi/blueprints/tooltool/grooming.py]: fixed, thanks

**Pull request merged**